### PR TITLE
[spot] Suppress unnecessary `spot launch` filemount translation logging.

### PR DESCRIPTION
**Pull request by concretevitamin**
Tested:

Should not print:
- [x]    `sky spot launch -n dbg ''`
- [x]    `/data.txt: s3://...`

Should print:
- [ ] Case
```
/data.txt:
    source: ~/data.txt
```
cannot test due to #1226
- [x] /data.txt: ~/data.txt
- [x] Case:  workdir exists

### CODE REVIEW

File: sky/execution.py [hunk 1]
```diff
@@ -426,6 +426,11 @@ def spot_launch(
 
 
 def _translate_local_file_mounts(task: task_lib.Task) -> task_lib.Task:
+    """Translates any local->VM mounts into Storage->VM mounts.
```
Comment (Michaelvll) [inline sky/execution.py:429]: Nice!

File: sky/execution.py [hunk 2]
```diff
@@ -533,8 +539,9 @@ def _translate_local_file_mounts(task: task_lib.Task) -> task_lib.Task:
     # Upload the local source to a bucket. The task will not be executed
     # locally, so we need to upload the files/folders to the bucket manually
     # here before sending the task to the remote spot controller.
-    logger.info(f'{colorama.Fore.YELLOW}Uploading sources to cloud storage.'
-                f'{colorama.Style.RESET_ALL} See sky storage ls')
+    if new_storage_mounts:
+        logger.info(f'{colorama.Fore.YELLOW}Uploading sources to cloud storage.'
+                    f'{colorama.Style.RESET_ALL} See: sky storage ls')
```
Comment (Michaelvll) [inline sky/execution.py:544]: This logging does not depend on the `new_storage_mounts`, as even without translation, users can still specify sky storages with source paths.

**Review by Michaelvll — APPROVED** [review]
Great catch! Thank you for fixing the logs @concretevitamin!

File: sky/execution.py (hunk 2 above)
Comment (concretevitamin) [reply sky/execution.py]: Got it, I totally missed this due to the function name. Changed the function name a bit to avoid the confusion. Wdyt?

**Review by Michaelvll — APPROVED** [review]
LGTM. Thanks @concretevitamin !

**Pull request merged**